l1-contracts/test/foundry/unit/concrete/Bridgehub/experimental_bridge.t.sol

```diff
@@ -0,0 +1,744 @@
+    DummySharedBridge mockSharedBridge;
+    TestnetERC20Token testToken;
+    //UtilsFacet utils;
+
+    function setUp() public {
+        bridgeHub = new Bridgehub();
+        bridgeOwner = makeAddr("BRIDGE_OWNER");
+        mockSTM = new DummyStateTransitionManagerWBH(address(bridgeHub));
+        mockChainContract = new DummyStateTransition(address(bridgeHub));
+        mockSharedBridge = new DummySharedBridge(keccak256("0xabc"));
+        testToken = new TestnetERC20Token("ZKSTT", "ZkSync Test Token", 18);
+
+        // test if the ownership of the bridgeHub is set correctly or not
+        address defaultOwner = bridgeHub.owner();
+
+        // The defaultOwner should be the same as this contract address, since this is the one deploying the bridgehub contract
+        assertEq(defaultOwner, address(this));
+
+        // Now, the `reentrancyGuardInitializer` should prevent anyone from calling `initialize` since we have called the constructor of the contract
+        // @follow-up Is this the intended behavior? @Vlad @kalman
+        vm.expectRevert(bytes("1B"));
+        bridgeHub.initialize(bridgeOwner);
+
+        vm.store(address(mockChainContract), 0x8e94fed44239eb2314ab7a406345e6c5a8f0ccedf3b600de3d004e672c33abf4, bytes32(uint256(1)));  
+        bytes32 bridgehubLocation = bytes32(uint256(36));
+        vm.store(address(mockChainContract), bridgehubLocation, bytes32(uint256(uint160(address(bridgeHub)))));
+        bytes32 baseTokenGasPriceNominatorLocation = bytes32(uint256(40));
+        vm.store(address(mockChainContract), baseTokenGasPriceNominatorLocation, bytes32(uint256(1)));
+        bytes32 baseTokenGasPriceDenominatorLocation = bytes32(uint256(41));
+        vm.store(address(mockChainContract), baseTokenGasPriceDenominatorLocation, bytes32(uint256(1)));
+        // The ownership can only be transfered by the current owner to a new owner via the two-step approach
+        
+        // Default owner calls transferOwnership
+        bridgeHub.transferOwnership(bridgeOwner);
+
+        // bridgeOwner calls acceptOwnership
+        vm.prank(bridgeOwner);
+        bridgeHub.acceptOwnership();
+
+        // Ownership should have changed
+        assertEq(bridgeHub.owner(), bridgeOwner);
+    }
+
+    function test_ownerCanSetDeployer(address randomCaller, address randomDeployer) public {
+        /**
+            Case I: A random address tries to call the `setDeployer` method and the call fails with the error: `Ownable: caller is not the owner`
+            Case II: The bridgeHub owner calls the `setDeployer` method on `randomDeployer` and it becomes the deployer
+        */
+        
+        if(randomCaller != bridgeHub.owner()) {
+            vm.prank(randomCaller);
+            vm.expectRevert(bytes("Ownable: caller is not the owner"));
+            bridgeHub.setDeployer(randomDeployer);
+
+            // The deployer shouldn't have changed.
+            assertEq(address(0), bridgeHub.deployer());
+        } else { 
```

Comment:
I doubt that this will ever be a target of the Foundry fuzzer. Shall you create a separate `unit` test for this?

Reply:
I mean, the case should of course be reached if we do infinite fuzz runs :P

But since we cannot do that, I have broken down this unit test into multiple unit tests.